### ISSUE TITLE
Error when calling vanilla or Forge classes from dependency library in kotlin environment

### ISSUE THREAD

**Issue by Hiiragi283** [GTNewHorizons/RetroFuturaGradle#51]
Example: 
```kotlin
val stack: ItemStack = ItemStack(Items.IRON_INGOT)
crafttweaker.api.minecraft.CraftTweakerMC.getItemStack(stack) <-
```

Error Message: `Cannot access class 'net.minecraft.item.ItemStack'. Check your module classpath for missing or conflicting dependencies`

**Comment by Hummel009** [comment]
I'm facing the same problem. I fixed it temporarily by putting the dependency source code in the build/rfg/minecraft-src folder. However, this is a bad way, because the decompiled dependency source contains errors.

Maybe you managed to solve this problem in a better way during these 2 months?

**Comment by slava110** [comment]
Same problem. Wasn't happening before to me IIRC, I've been using RFG earlier with Kotlin without problems :I

**Comment by Hiiragi283** [comment]
@slava110 
oh really?
i didn’t know that solution

**Comment by eigenraven** [comment]
Have a look at <https://github.com/GTNewHorizons/MX-Random/blob/master/addon.gradle> - it builds with latest RFG using the GTNH buildscript without issues and is a kotlin mod.

**Comment by eigenraven** [comment]
And if that doesn't help, please send a code sample or repository I can actually test to see what's going on

**Comment by Hiiragi283** [comment]
> Have a look at https://github.com/GTNewHorizons/MX-Random/blob/master/addon.gradle - it builds with latest RFG using the GTNH buildscript without issues and is a kotlin mod.

well… is that project using RFG? i couldn’t find any reference in *.gradle files

**Comment by eigenraven** [comment]
Yes, almost all GTNH projects use RFG. We now switched to GTNHGradle which uses RFG as a dependency, but you can look at the old buildscript here <https://github.com/GTNewHorizons/MX-Random/blob/8a6903997e9c34e991f607554d179367c3d8d3b6/build.gradle>

**Comment by Hiiragi283** [comment]
Can i use GTNHGradle for MC1.12.2?

**Comment by slava110** [comment]
@eigenraven I've created minimal reproducible example here:
https://github.com/slava110/RFGKotlinProblem
I'm depending on `CreativeCore` mod with `rfg.deobf` on it [here](https://github.com/slava110/RFGKotlinProblem/blob/81192ee99a62c96040603ab7f5ff090f8344afff/build.gradle.kts#L30)
After initial setup you will (probably) be able to see error in [ExampleMod.kt](https://github.com/slava110/RFGKotlinProblem/blob/81192ee99a62c96040603ab7f5ff090f8344afff/src/main/kotlin/com/example/examplemod/ExampleMod.kt#L10) file like this one:
![image](https://github.com/GTNewHorizons/RetroFuturaGradle/assets/26705162/fde88a10-d542-409a-8ad1-e7e015ba2a78)

**Comment by shorinami** [comment]
Any updates on this, found any workarounds?

**Comment by shorinami** [comment]
> Yes, almost all GTNH projects use RFG. We now switched to GTNHGradle which uses RFG as a dependency, but you can look at the old buildscript here https://github.com/GTNewHorizons/MX-Random/blob/8a6903997e9c34e991f607554d179367c3d8d3b6/build.gradle

That project has same problem 
![image](https://github.com/GTNewHorizons/RetroFuturaGradle/assets/137571219/e23067e0-843f-4d22-93c2-6c6c1addab62)

**Comment by Glease** [comment]
Try run `gradlew setupDecompWorkspace`. I got the same issue on your test repo before running that command. The issue gies away afterwards.

**Comment by shorinami** [comment]
> Try run `gradlew setupDecompWorkspace`. I got the same issue on your test repo before running that command. The issue gies away afterwards.

I did run it, does not help

**Comment by Glease** [comment]
If anything,
* I bumped gradle version in gradlew setting to 8.6 
* I'm running gradle with java 17
* I'm using intellij 2023.3.x, with RFG enabled version of minecraftdev plugin

I don't think any of these would practically matter, but that's probably the only outstanding points

also can you build the project by running gradlew build?

**Comment by shorinami** [comment]
> If anything,
> 
> * I bumped gradle version in gradlew setting to 8.6 
> 
> * I'm running gradle with java 17
> 
> * I'm using intellij 2023.3.x, with RFG enabled version of minecraftdev plugin
> 
> 
> 
> I don't think any of these would practically matter, but that's probably the only outstanding points
> 
> 
> 
> also can you build the project by running gradlew build?

Yes, the project builds just fine, also if I access the library from java, it works just fine (without no warnings/errors), so it is related to how IDEA handles Kotlin I think. Will check the other stuff and let you know if that helps

**Comment by shorinami** [comment]
> If anything,
> 
> * I bumped gradle version in gradlew setting to 8.6
> * I'm running gradle with java 17
> * I'm using intellij 2023.3.x, with RFG enabled version of minecraftdev plugin
> 
> I don't think any of these would practically matter, but that's probably the only outstanding points
> 
> also can you build the project by running gradlew build?

Did not help, rolled back to 2023.3.6, java 17, gradle 8.6.
Only thing I did not understand fully, what do you mean by "RFG enabled"?

**Comment by Glease** [comment]
https://github.com/eigenraven/MinecraftDev as opposed to the one you would find on plugin marketplace

**Comment by Glease** [comment]
How about gradlew clean and delete .idea folder (while IDE is closed of course)?

**Comment by shorinami** [comment]
> How about gradlew clean and delete .idea folder (while IDE is closed of course)?

Yeah, same setup, same problem:
![image](https://github.com/GTNewHorizons/RetroFuturaGradle/assets/137571219/c627f636-a0b0-430e-87df-2f3373245246)

**Comment by Hummel009** [comment]
Still have the same problem. I tried different Kotlin (1.8.22 — 1.9.23) + Gradle (8.2 — 8.7) + system-wide JDK (1.8 — 21), and nothing helped.

The problem can be solved in bad way, to decompile the .jar dependency and to put its classes into rfg/minecraft-src. But you know, the quality of decompiled dependencies is low.

**Comment by shorinami** [comment]
@Hummel009 @Glease
So I figured out an interesting fix for that.
After turning on the Kotlin K2 Mode in Kotlin language settings the issue is gone.

![image](https://github.com/GTNewHorizons/RetroFuturaGradle/assets/137571219/155a5abb-0835-4077-9d09-a22cfd61ee6f)

**Comment by shorinami** [comment]
I also filled out an issue on the JetBrains YouTrack as it might be an issue on their end: https://youtrack.jetbrains.com/issue/IDEA-352220/Cannot-access-class-net.minecraftforge.fml.common.event.FMLServerStartingEvent.-Check-your-module-classpath-for-missing-or

**Comment by Hummel009** [comment]
Yes, compiler K2 works perfect with retrofuturagradle. Thanks!

**Comment by Hummel009** [comment]
At this point, Kotlin 2.0 is out. They say that now we don't need to additionally enable K2 mode, as it's the default from now on. However, the RetroFuturaGradle error comes back as soon as I uncheck K2 mode in IDEA Settings.

This is very strange, because the checkmark should now mean nothing. Nevertheless, I would like to uncheck it, because then half of the plugins and even the build script stops working. But the RetroFuturaGradle starts working perfect. You have to choose what will work and what will be broken.
